projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts

```diff
@@ -163,20 +162,19 @@ export class TopologyDataSourceModel extends GraphQlDataSourceModel<TopologyData
   }
 
   private getTopologyFilters(filters: GraphQlFilter[]): TopologyFilters {
-    const edgeFilterFields = this.edgeFilterConfig?.fields ?? [];
-    const edgeFilters: GraphQlFilter[] = [];
+    const edgeFilterFields = this.edgeFilterConfig?.entityConfigs.flatMap(info => info.fields) ?? [];
+    const edgeFilters: Map<string, GraphQlFilter[]> = new Map();
     const nodeFilters: GraphQlFilter[] = [];
 
-    const isFieldFilter = (gqlFilter: GraphQlFilter): gqlFilter is GraphQlFieldFilter =>
-      'keyOrExpression' in gqlFilter && 'operator' in gqlFilter && 'value' in gqlFilter;
-
     filters.forEach(gqlFilter => {
       // Edge filter only supported for `GraphQlFieldFilter` for now
-      if (
-        isFieldFilter(gqlFilter) &&
-        edgeFilterFields.includes(this.getFieldFromExpression(gqlFilter.keyOrExpression))
-      ) {
-        edgeFilters.push(gqlFilter);
+      const field = this.isFieldFilter(gqlFilter) ? this.getFieldFromExpression(gqlFilter.keyOrExpression) : '';
```

Comment:
instead of just skipping past by assuming `''` as default value, can we do something better?

Do we expect non `GraphqlFieldFilters` here? If yes, is it by design or is it an unexpected scenario? If its the former, this is fine. If its latter, we should throw/log an appropriate error.

Reply:
We have an interface called `GraphQlFilter` that gets implemented and `GraphQlFieldFilter` is one of them. Now the data source and the query handler accept `GraphQlFilter`.  This logic only handles the edge filters and it only considers the field-filters for now. All the other filters go directly inside the node filters instead so I think we're good on that front.

Reply:
replace it with undefined?

Reply:
Done